hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/AvroKafkaSource.java

```diff
@@ -60,7 +66,7 @@ public AvroKafkaSource(TypedProperties props, JavaSparkContext sparkContext, Spa
     super(props, sparkContext, sparkSession, schemaProvider);
 
     props.put(NATIVE_KAFKA_KEY_DESERIALIZER_PROP, StringDeserializer.class);
-    String deserializerClassName = props.getString(DataSourceWriteOptions.KAFKA_AVRO_VALUE_DESERIALIZER(), "");
+    String deserializerClassName = props.getString(KAFKA_AVRO_VALUE_DESERIALIZER_OPT_KEY(), DEFAULT_KAFKA_AVRO_VALUE_DESERIALIZER_OPT_VAL());
```

Comment:
we need to be careful in line 71. since we have updated the default value, we may never go into if condition of line 71. Can you fix that please.

Reply:
Hi, i would prefer to leave the block in - even if we turn the feature on by default. Doing so we have the possibility to turn the feature explicitly off (`hoodie.deltastreamer.source.kafka.value.deserializer.class=`) if it causes any problems

Reply:
I am not sure I get you. Can you clarify please. 
As of now, it looks fine, since the default value is empty string. I am talking about when we make the new deser class as default. We can come back to revisit this when we fix the default value.

Reply:
I agree with you.
But having a way to disable the feature (setting `(hoodie.deltastreamer.source.kafka.value.deserializer.class` to empty) is always a good idea. But for this to work we need to leave in the Codeblock checking for an empty string.
So I would suggest to leave it in regardless of the default value.

Reply:
While rebasing i found a way how we can remove this isEmpty() check. Take a look at the current patch.